src/ex4/server/services/item_manager.js

```diff
@@ -1 +1,171 @@
-// The ItemManager should go here. Remember that you have to export it.
\ No newline at end of file
+const {
+  POKEMON_WITH_ID_NOT_FOUND,
+  NOT_A_POKEMON,
+  FAILED_TO_FETCH,
+} = require("./globalConsts/GlobalConstants.js");
+const fs = require("fs");
+const PokemonClient = require("../clients/pokemon_client.js");
+const path = require("path");
+
+class ItemManager {
+  constructor() {
+    this.pokedex = PokemonClient;
+    this.tasksFile = "../DB/tasks.json";
+    this.cacheFile = "../DB/pokemonsCache.json";
+    try {
+      this.tasks = this.getTasks();
+    } catch (e) {
+      this.tasks = [];
+    }
+  }
+
+  getTasksLength() {
+    return this.tasks.length;
+  }
+
+  isInputSetOfPokemonIDs(input) {
+    const regex = /^[0-9,]+$/;
+    return regex.test(input);
+  }
+
+  async addTask(taskInput, isCompleted) {
+    if (
+      this.isInputSetOfPokemonIDs(taskInput) ||
+      this.pokedex.isPokemonNamesOnly(taskInput)
+    ) {
+      return await this.addCatchPokemonTask(taskInput);
+    } else {
+      const isAdded = this.addToFile(taskInput, isCompleted);
+      return isAdded;
+    }
+  }
+
+  async addCatchPokemonTask(input) {
+    //check if in cache first
+    let response = null;
+    const cache = this.getCache();
+    const cacheItem = cache.find((item) => item.input === input);
```

Comment:
if you always find the items from the cache by their input, maybe it should be the key? you can have an object instead of an array for easier (and more efficient) access